File: Client/app/src/main/java/com/shortdrama/app/data/model/Drama.kt
package com.shortdrama.app.data.model

/**
 *
 * 后端路由: GET /api/dramas（列表） / GET /api/dramas/:id（详情）
 *
 * 列表接口返回的 episodes = []（后端 omitempty，Gson 使用默认值）
 * 详情接口返回的 episodes = 该剧所有剧集
 */
data class Drama(
    val id: Long,
    val title: String,
    val description: String? = null,
    val cover_url: String? = null,
    val created_at: String? = null,
    val episodes: List<Episode> = emptyList()
)

File: Backend/internal/model/drama.go
package model

type Drama struct {
	ID          uint64    `json:"id" gorm:"primaryKey;autoIncrement"`
	Title       string    `json:"title" gorm:"size:255;not null"`
	Description string    `json:"description"`
	CoverURL    string    `json:"cover_url" gorm:"size:500"`
	Episodes    []Episode `json:"episodes,omitempty" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

func (Drama) TableName() string {
	return "drama"
}
